### PR TITLE
chore: add gitattributes file

### PR DESCRIPTION
**Pull request by jthegedus**
# Summary

This seems relevant now we have users on Windows operating systems (I am one on occasion)

---

~I happy to trim some of these items as I copied from an old project. It seems https://gitattributes.io is down now so I can't grab a new default version.~

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+*.od       text
+*.onlydata text
+*.php      text
+*.pl       text
+*.py       text
+*.rb       text
+*.sass     text
+*.scm      text
+*.scss     text
+*.sh       text eol=lf
```
Comment (Stratus3D) [inline .gitattributes:39]: Many of our files have the extension `.bash`, and one has the extension `.fish`. Do those extensions need to be added here as well?

Comment (jthegedus) [reply .gitattributes:39]: Yes, I'll update the file extensions we have in our repo

Comment (jthegedus) [reply .gitattributes:22]: L14 covers anything not explicitly defined here.

Comment (jthegedus) [reply .gitattributes]: `*.sh       text` will default to LF in the git object tree, but not in working directory. So check-in will always be LF and checkout will be system dependent.

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+*.html     text
+*.js       text
+*.sh       text
+*.bash     text
+*.fish     text
+#### asdf/bin/* explicitly define as text
+*asdf      text
+*asdf-exec text
+*_asdf     text
+#### asdf/test/fixtures/* should be covered by * text=auto on L14
```
Comment (jthegedus) [inline .gitattributes:29]: Documentation for an event where `* text=auto` does not pick up any of these files as `text` files. It's possible, but extremely unlikely.

File: .gitattributes [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+## SOURCE CODE
+*.bats     text
+*.css      text
+*.htm      text
+*.html     text
+*.js       text
+*.sh       text
+*.bash     text
+*.fish     text
+#### asdf/bin/* explicitly define as text
```
Comment (jthegedus) [inline .gitattributes:25]: This section is out of an abundance of caution re `* text=auto` as noted in comment on L29

**Pull request merged**